pkg/ebpf/compiler/compiler.cpp

```diff
@@ -0,0 +1,273 @@
+#include "compiler.h"
+#include "files.h"
+
+#include <iostream>
+#include <llvm/IR/LegacyPassManager.h>
+#include <llvm/Support/TargetSelect.h>
+#include <clang/CodeGen/CodeGenAction.h>
+#include <clang/Driver/Job.h>
+#include <clang/Driver/Tool.h>
+#include <clang/Frontend/CompilerInstance.h>
+#include <clang/Lex/PreprocessorOptions.h>
+
+bool ClangCompiler::llvmInitialized = false;
+
+enum Architecture { PPC, PPCLE, S390X, ARM64, X86 };
+
+std::map<std::string, std::unique_ptr<llvm::MemoryBuffer>> ClangCompiler::remapped_files_;
+
+ClangCompiler::ClangCompiler(const char *name) :
+    llvmContext(new llvm::LLVMContext),
```

Comment:
You can use `std::make_unique<llvm::LLVMContext>()` to avoid  a memory leak in case of an exception. See https://herbsutter.com/gotw/_102/ for a detailed explanation.